File: Portfolio/Portfolio.Blazor/Data/Entities/CreditEntryTranslation.cs
namespace Portfolio.Blazor.Data.Entities;

public sealed class CreditEntryTranslation
{
    public int Id { get; set; }
    public int CreditEntryId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public CreditEntry? CreditEntry { get; set; }
}

File: src/laravel/app/Models/CreditEntryTranslation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class CreditEntryTranslation extends Model
{
    use HasFactory;

    protected $fillable = ['credit_entry_id', 'locale', 'name', 'description'];

    /**
     * @return BelongsTo<CreditEntry, $this>
     */
    public function creditEntry(): BelongsTo
    {
        return $this->belongsTo(CreditEntry::class);
    }
}
